From ffdce4d9abc0c9b1ab233777489c8e7406c737a2 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Tue, 19 Sep 2023 00:57:29 +0200
Subject: [PATCH 005/102] enable LIST_HARDENED by default

Signed-off-by: Levente Polyak <levente@leventepolyak.net>
---
 security/Kconfig.hardening | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index c9d5ca3d8d08..8d5213ef63bd 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -283,6 +283,7 @@ menu "Hardening of kernel data structures"
 
 config LIST_HARDENED
 	bool "Check integrity of linked list manipulation"
+	default y
 	help
 	  Minimal integrity checking in the linked-list manipulation routines
 	  to catch memory corruptions that are not guaranteed to result in an
-- 
2.51.2

