From 1d62382c2b03c1981892b1e2abb91223afa58721 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sat, 27 May 2017 07:29:45 -0400
Subject: [PATCH 024/102] disable PROC_VMCORE by default

---
 fs/proc/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/proc/Kconfig b/fs/proc/Kconfig
index d80a1431ef7b..d84e627cc1fa 100644
--- a/fs/proc/Kconfig
+++ b/fs/proc/Kconfig
@@ -41,7 +41,6 @@ config PROC_KCORE
 config PROC_VMCORE
 	bool "/proc/vmcore support"
 	depends on PROC_FS && CRASH_DUMP
-	default y
 	help
 	  Exports the dump image of crashed kernel in ELF format.
 
-- 
2.51.2

