From 85c86cb796d94c5fec4b99dacec381dced50bff2 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Thu, 19 Sep 2019 19:03:01 +0200
Subject: [PATCH 040/102] enable INIT_ON_FREE_DEFAULT_ON by default

---
 security/Kconfig.hardening | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/Kconfig.hardening b/security/Kconfig.hardening
index 5f6c924a1c88..121eac9a89f2 100644
--- a/security/Kconfig.hardening
+++ b/security/Kconfig.hardening
@@ -240,6 +240,7 @@ config INIT_ON_ALLOC_DEFAULT_ON
 
 config INIT_ON_FREE_DEFAULT_ON
 	bool "Enable heap memory zeroing on free by default"
+	default yes
 	depends on !KMSAN
 	help
 	  This has the effect of setting "init_on_free=1" on the kernel
-- 
2.51.2

