From 7e1e83fd2a5b90429d1f1ad22376725b350243c9 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Mon, 1 May 2023 23:20:43 +0200
Subject: [PATCH 047/102] kconfig: disable LEGACY_TIOCSTI by default

---
 drivers/tty/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/tty/Kconfig b/drivers/tty/Kconfig
index 03885416cd43..a4b0a144ec58 100644
--- a/drivers/tty/Kconfig
+++ b/drivers/tty/Kconfig
@@ -145,7 +145,6 @@ config LEGACY_PTY_COUNT
 
 config LEGACY_TIOCSTI
 	bool "Allow legacy TIOCSTI usage"
-	default y
 	help
 	  Historically the kernel has allowed TIOCSTI, which will push
 	  characters into a controlling TTY. This continues to be used
-- 
2.51.2

