From 775eba4ebf809164a0b53792e1bb6f13398f717f Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Tue, 12 Dec 2023 18:35:05 +0100
Subject: [PATCH 048/102] mm/slab: enable RANDOM_KMALLOC_CACHES by default

---
 mm/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/Kconfig b/mm/Kconfig
index ee6e9a7a589c..ce2cf583cfd1 100644
--- a/mm/Kconfig
+++ b/mm/Kconfig
@@ -329,7 +329,7 @@ config SLUB_CPU_PARTIAL
 	  Typically one would choose no for a realtime system.
 
 config RANDOM_KMALLOC_CACHES
-	default n
+	default y
 	depends on !SLUB_TINY
 	bool "Randomize slab caches for normal kmalloc"
 	help
-- 
2.51.2

