From c68a4f857338ecac3d6e34764b58eb1f3fc8d92a Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Sun, 7 May 2017 00:28:23 -0400
Subject: [PATCH 049/102] add __read_only for non-init related usage

---
 include/linux/cache.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/cache.h b/include/linux/cache.h
index ca2a05682a54..2fda42e7a36b 100644
--- a/include/linux/cache.h
+++ b/include/linux/cache.h
@@ -63,6 +63,8 @@
 #define __ro_after_init __section(".data..ro_after_init")
 #endif
 
+#define __read_only __ro_after_init
+
 #ifndef ____cacheline_aligned
 #define ____cacheline_aligned __attribute__((__aligned__(SMP_CACHE_BYTES)))
 #endif
-- 
2.51.2

