From 30cd4be9cc3c3bb11297d41e7427cd0ca6169722 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@ariadne.space>
Date: Wed, 10 Jul 2024 14:51:47 -0700
Subject: [PATCH 09/29] sysctl: wire up PaX softmode setting

Signed-off-by: Ariadne Conill <ariadne@ariadne.space>
---
 kernel/sysctl.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/kernel/sysctl.c b/kernel/sysctl.c
index 79e6cb1d5c48..e28466b47720 100644
--- a/kernel/sysctl.c
+++ b/kernel/sysctl.c
@@ -1610,6 +1610,18 @@ int proc_do_static_key(const struct ctl_table *table, int write,
 	return ret;
 }
 
+#ifdef CONFIG_OPENPAX_SOFTMODE
+static struct ctl_table pax_table[] = {
+	{
+		.procname       = "softmode",
+		.data           = &pax_softmode,
+		.maxlen         = sizeof(int),
+		.mode           = 0600,
+		.proc_handler   = proc_dointvec,
+	},
+};
+#endif
+
 static struct ctl_table kern_table[] = {
 	{
 		.procname	= "panic",
@@ -2240,6 +2252,9 @@ int __init sysctl_init_bases(void)
 {
 	register_sysctl_init("kernel", kern_table);
 	register_sysctl_init("vm", vm_table);
+#ifdef CONFIG_OPENPAX_SOFTMODE
+	register_sysctl_init("kernel/pax", pax_table);
+#endif
 
 	return 0;
 }
-- 
2.51.2

