From 3cdd883913e5e293c3e5461429e9c8db5604463f Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@ariadne.space>
Date: Sun, 21 Jul 2024 18:40:14 -0700
Subject: [PATCH 21/29] security: openpax: add OPENPAX_EMUTRAMP_DEFAULT config
 option

Signed-off-by: Ariadne Conill <ariadne@ariadne.space>
---
 security/Kconfig.openpax | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/security/Kconfig.openpax b/security/Kconfig.openpax
index d8ee4bc570b6..76ee145094d9 100644
--- a/security/Kconfig.openpax
+++ b/security/Kconfig.openpax
@@ -72,4 +72,18 @@ config OPENPAX_EMUTRAMP
 	
 	  Most likely, you should say 'y' here.
 
+config OPENPAX_EMUTRAMP_DEFAULT
+	bool "Enable trampoline emulation by default"
+	depends on OPENPAX_EMUTRAMP
+	default y
+	help
+	  Enabling this option allows programs which require
+	  trampolines to be emulated to continue working by default.
+
+	  Otherwise, the emulation flag must be enabled in a binary's
+	  PaX marking, e.g. with paxmark -E <binary>.
+
+	  If you do not say 'y' here, you will have to manually mark
+	  all programs which require trampoline emulation.
+
 endmenu
-- 
2.51.2

